### PR TITLE
move pkg/kubelet/apis/well_known_labels.go to staging/src/k8s.io/api/core/v1/

### PR DESCRIPTION
**Pull request by dims**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

part of #69585 to remove cloudprovider dependencies to pkg/kubelet/apis

Co-Authored-By: Weibin Lin <linweibin1@huawei.com>

Change-Id: I163b2f2833e6b8767f72e2c815dcacd0f4e504ea

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by andrewsykim** [comment]
LGTM :)

**Comment by andrewsykim** [comment]
/lgtm

Thanks @dims!

**Review by neolit123 — APPROVED** [review]
thanks!

/priority important-longterm
/approve

File: pkg/kubelet/apis/well_known_labels.go
Comment (thockin) [inline pkg/kubelet/apis/well_known_labels.go]: why not these?

File: pkg/kubelet/apis/well_known_labels.go
Comment (thockin) [inline pkg/kubelet/apis/well_known_labels.go]: Shouldn't `v1.LabelZoneRegion` looks like a brain-o.  Shouldn't it be `v1.LabelRegionFailureDomain`  to be parallel?

Comment (thockin) [reply pkg/kubelet/apis/well_known_labels.go]: I really really don't like that we have this codified as an API.  It's such a provider-specific detail.

File: staging/src/k8s.io/api/core/v1/well_known_labels.go
Comment (thockin) [inline staging/src/k8s.io/api/core/v1/well_known_labels.go]: I am trying to understand what this is for, it really seems like it should not be exaposed but be a better API for callers who need to parse thusly formatted strings?

**Review by thockin — COMMENTED** [review]
Some of these comments are for later or larger consideration.  Some seem wrong now.

File: pkg/kubelet/apis/well_known_labels.go
Comment (andrewsykim) [reply pkg/kubelet/apis/well_known_labels.go]: Likely more context here https://github.com/kubernetes/kubernetes/pull/68267, but the idea was to prevent usage of these labels until kubelet/controllers were applying these labels (in a backwards compatible way). I don't see harm in exporting them now as long as there is a follow up to apply the GA labels where needed (a bunch of cloud specific controllers need this update). @liggitt may have more context here.

Comment (andrewsykim) [reply pkg/kubelet/apis/well_known_labels.go]: There probably needs to be a KEP for how we're going to deprecate the beta labels going forward though?

Comment (liggitt) [reply pkg/kubelet/apis/well_known_labels.go]: > There probably needs to be a KEP for how we're going to deprecate the beta labels going forward though?

Yup. See https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/20190130-node-os-arch-labels.md as a model.

Comment (andrewsykim) [reply pkg/kubelet/apis/well_known_labels.go]: Thanks @liggitt, I can follow up to write a KEP on this for zone/region, instance types and other cloud-specific ones that need to be addressed. Will leave it up to @thockin and @dims if they want to keep these here or until after the KEP is approved.

Comment (dims) [reply pkg/kubelet/apis/well_known_labels.go]: i feel this PR is just for code re-organization ( moving things around ). yes +1 to a follow up KEP for other things

File: staging/src/k8s.io/api/core/v1/well_known_labels.go
Comment (liggitt) [reply staging/src/k8s.io/api/core/v1/well_known_labels.go]: Agree. The actual label keys make some sense. This, not as much.

File: pkg/kubelet/apis/well_known_labels.go
Comment (dims) [reply pkg/kubelet/apis/well_known_labels.go]: Ack fixing this

File: staging/src/k8s.io/api/core/v1/well_known_labels.go
Comment (dims) [reply staging/src/k8s.io/api/core/v1/well_known_labels.go]: ack. moving LabelMultiZoneDelimiter to a better spot

**Comment by dims** [comment]
@thockin @liggitt this should be ready once we get past the e2e-gce flake

File: pkg/volume/gcepd/attacher_test.go
Comment (cheftako) [inline pkg/volume/gcepd/attacher_test.go]: I realize gce already depends on volumeutil, so it hasn't made things significantly worse. However volumeutil is not part of staging. So we have clearly moved something (LabelMultiZoneDelimiter) which needed to be moved into staging somewhere other than staging. Are we planning to either move it again or to move volumeutil into staging?

**Review by thockin — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**
